src/librbd/internal.cc

```diff
@@ -1913,6 +1921,20 @@ int mirror_image_disable_internal(ImageCtx *ictx, bool force) {
       unknown_format = false;
       id = ictx->id;
 
+      if (force) {
+        C_SaferCond ctx;
+        ictx->exclusive_lock->shut_down(&ctx);
```

Comment:
`exclusive_lock` might not be initialized yet -- perhaps move this down to be within the block near line 1939.  Probably should be holding a write lock against the `owner_lock` as well.